primitives/forest-verifier/src/lib.rs

```diff
@@ -204,6 +204,64 @@ where
             }
         }
 
-        return Ok(Vec::from_iter(proven_keys));
+        return Ok(proven_keys);
+    }
+}
+
+// TODO: rename to TrieDeltaApplier
```

Comment:
Was this left dangling? Maybe replace for a doc comment con this implementation?